src/System.Buffers.ReaderWriter/System/Buffers/Reader/BufferReader.cs

```diff
@@ -156,12 +156,7 @@ private void GetNextSpan()
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public void Advance(int count)
         {
-            if (count == 0)
-            {
-                return;
-            }
-
-            if (count < 0 || End)
```

Comment:
FYI, @JeremyKuhne, I am relying on the existing tests to verify correctness here.

As far as I can reason about this, we can let `count == 0` pass through the rest of the code and nothing changes (effectively equivalent to just returning).

Similarly, we can let `End` be true and not check since the `AdvanceToNextSpan(int count)` already does the check and throws the exact exception anyway. Let me know if you agree.

If `count > 0` and `End` is true, `CurrentSpanIndex` must equal `CurrentSpan.Length` and hence we will always fall to the `AdvanceToNextSpan` code path. Is my reasoning correct?

Reply:
> Is my reasoning correct?

I think so? Just make sure we've got coverage in place. Want to make sure we cover the return count on the APIs and the situations you're discussing here.